kingpin/actors/rightscale/deployment.py

```diff
@@ -119,13 +116,119 @@ def _execute(self):
             self._client._client.deployments, params)
 
 
+class Clone(DeploymentBaseActor):
+
+    """Clones a RightScale deployment.
+
+    **Options**
+
+    :clone:
+      The name of the deployment to clone.
+
+    :name:
+      The name of the deployment to create.
+
+    :description:
+       The description of the new deployment.
+       (*optional*)
+
+    :server_tag_scope:
+       The routing scope for tags for servers in the deployment.
+       Can be 'deployment' or 'account'
+       (*optional*, default: deployment)
+    """
+
+    all_options = {
+        'clone': (str, REQUIRED, 'The name of the deployment to clone.'),
+        'name': (str, REQUIRED, 'The name of the deployment to create.'),
+        'description': (
+            str, '',
+            'The description of the deployment to be cloned.'),
+        'server_tag_scope': (
+            str, '',
+            'The routing scope for tags for servers in the deployment.'),
+        'delete_servers': (
+            bool, False, 'Delete servers and arrays after cloning')
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Validate the user-supplied parameters at instantiation time."""
+
+        super(Clone, self).__init__(*args, **kwargs)
+
+        allowed_scopes = ('deployment', 'account', '')
+
+        scope = self.option('server_tag_scope')
+        if scope not in allowed_scopes:
+            raise exceptions.InvalidOptions(
+                'server_tag_scope "%s" is not one of: %s' % (scope,
+                                                             allowed_scopes))
+
+    @gen.coroutine
+    def _execute(self):
+
+        dep = yield self._find_deployment(self.option('clone'))
+        if not dep:
+            raise exceptions.InvalidOptions(
+                'Deployment "%s" does not exist.' % self.option('clone'))
+        self.log.info('Found "%s" to clone.' % self.option('clone'))
+
+        newdep = yield self._find_deployment(self.option('name'))
```

Comment:
I know its a little bit silly, but you can group lines 170 and 176 so that the calls happen asynchronously. :)